dashboard/app/models/ability.rb

```diff
@@ -268,6 +268,7 @@ def initialize(user)
 
       if SingleUserExperiment.enabled?(user: user, experiment_name: 'ai-differentiation') && user.teacher?
         can :chat_completion, :ai_diff
+        can :submit_feedback, AichatMessage
```

Comment:
You might be able to get rid of the `set_aichat_message` method in the controller if you do this. Pushes the authorization step into the CanCan system. Give it a try?

```suggestion
        can :submit_feedback, AichatMessage, aichat_thread: {user: user}
```

Reply:
I tried this, and it was still able to add feedback on a message that the user didn't own. I'm not sure why but after trying to figure it out for hours tbh I kind of gave up

Reply:
Ok fair enough! Never mind then.